pkg/sql/plan/build_show.go

```diff
@@ -246,7 +246,11 @@ func buildShowTables(stmt *tree.ShowTables, ctx CompilerContext) (*Plan, error)
 		return nil, moerr.NewNoDB()
 	}
 	ddlType := plan.DataDefinition_SHOW_TABLES
-	sql := fmt.Sprintf("SELECT relname as Tables_in_%s FROM %s.mo_tables WHERE reldatabase = '%s' and relname != '%s'", dbName, MO_CATALOG_DB_NAME, dbName, "%!%mo_increment_columns")
+	var tableType string
+	if stmt.Full {
+		tableType = ", case relkind when 'v' then 'VIEW' else 'BASE TABLE' end as Table_type"
+	}
+	sql := fmt.Sprintf("SELECT relname as Tables_in_%s %s FROM %s.mo_tables WHERE reldatabase = '%s' and relname != '%s'", dbName, tableType, MO_CATALOG_DB_NAME, dbName, "%!%mo_increment_columns")
```

Comment:
This is subject to sql injection.

Reply:
> This is subject to sql injection.

bvt tests about this added. no sql injection problem here

Reply:
Of course there is sql injection here.   You are just not creative enough to come up with one.

There must be a quote_identify call, or using param binding.

Reply:
this patch is to support show full tables statement. 
sql injection problem will be fixed in https://github.com/matrixorigin/matrixone/issues/5476